R/modules/upload-documentation.R

```diff
@@ -0,0 +1,85 @@
+
+          h4("Study Description"),
+
+          p("The study description is an overview of the study and should include:"),
+
+          tags$ul(
+            tags$li("human studies", tags$ul(tags$li("how the data was obtained, as well as a summary description of the data, including study type (prospective cohort, case-control, or post-mortem), disease focus, inclusion/exclusion criteria, and number of participants or donors. For post mortem studies, include the brain bank name(s) and tissue(s) that were sampled."))),
+            tags$li("model studies", tags$ul(tags$li("where the models were generated, as well as a summary description of the model, including common name, genetic background, and a link to the strain datasheet, or datasheets if a cross between two strains.")))
+          ),
+
+          h4("Assay Description"),
+
+          p("The assay description should include a summary of ", tags$b("sample processing, data generation,"), " and ", tags$b("data processing.")),
+
+          #nolint end
+
+          # UI for getting the study name
+          get_study_ui(ns("doc_study")),
+
+          # File import
+          fileInput(ns("study_doc"),
+                    "Upload the study description file"),
+          fileInput(ns("assay_doc"),
+                    "Upload the assay description files",
+                    multiple = TRUE),
+
+          # Add an indicator feature to submit button
+          with_busy_indicator_ui(
+            actionButton(
+              ns("upload_docs"),
+              "Submit"
+            )
+          )
+  )
+}
+
+#' Server function for upload_documentation module
+#' @param input the input from shiny::callModule
+#' @param output the output from shiny::callModule
+#' @param session the session from shiny::callModule
+upload_documents <- function(input, output, session, parent_folder) {
+  # Create folder for upload
+  docs_folder <- synapser::Folder(name = "Documentation",
+                                  parent = parent_folder())
+  created_docs_folder <- synapser::synStore(docs_folder)
+
+  # Get the study name
+  study_name <- callModule(get_study, "doc_study")
+  doc_annots <- reactive({
+    list(study = study_name())
+  })
+
+  # Upload files to Synapse (after renaming them so they keep their original
+  # names)
+  observeEvent(input$upload_docs, {
+    if (!is.null(input$study_doc) || !is.null(input$assay_doc)) {
+      # When the button is clicked, wrap the code in the call to the
+      # indicator server function
+      with_busy_indicator_server("upload_docs", {
+        all_docs <- rbind(input$study_doc, input$assay_doc)
```

Comment:
If they have only provided study or assay docs (but not both) does this still work? I think it should but just want to double check. I think we should accept them submitting one or the other, since sometimes for existing studies they may not need to update the existing study description, only the assay description.

Reply:
This does work. I tested it (manually, of course), but I was surprised that it worked correctly, as well. They can upload either set of documents or both sets of documents. It makes the code far cleaner and gets the job done. Yay!